lib/Backend/Lower.cpp

```diff
@@ -4932,15 +4934,47 @@ Lowerer::LowerNewScObjArray(IR::Instr *newObjInstr)
         Assert(weakFuncRef);
     }
 
-    IR::Opnd *opndSrc1 = newObjInstr->UnlinkSrc1();
-    if (opndSrc1->IsImmediateOpnd())
+    IR::LabelInstr *labelDone = IR::LabelInstr::New(Js::OpCode::Label, func);
+    IR::Opnd *opndSrc = newObjInstr->GetSrc2();
+
+    Assert(opndSrc->IsSymOpnd());
+    StackSym *linkSym = opndSrc->AsSymOpnd()->m_sym->AsStackSym();
+    Assert(linkSym->IsSingleDef());
+    IR::Instr* argInstr = linkSym->GetInstrDef();
+    IR::Opnd *opndOfArrayCtor = argInstr->GetSrc1();
+    // Generate fast path only if it meets all the conditions:
+    // 1. 1st paramter is a constant
+    // 2. It is in range 0 <= parameter <= 8
+    // 3. It is the only parameter.
+    if (opndOfArrayCtor->IsIntConstOpnd() || opndOfArrayCtor->IsAddrOpnd()) // #1
```

Comment:
>(opndOfArrayCtor->IsIntConstOpnd() [](start = 7, length = 34)

since you are already doing this, you can directly call opndOfArrayCtor->GetImmediateValue ?